app/controllers/services_controller.rb

```diff
@@ -0,0 +1,49 @@
+class ServicesController < ApplicationController
+  ## EXAMPLE
+    # params = { packages: [{ dimensions: [25, 10, 15], weight: 500 }, { dimensions: [18, 30, 10], weight: 5000 }], origin: { state: "WA", city: "Seattle", zip: "98101" }, destination: { state: "IL", city: "Vernon Hills", zip: "60061" } }.to_json
+    # r = HTTParty.post("http://localhost:3000/ups/", headers: { 'Content-Type' => 'application/json' }, body: params)
+    # r = HTTParty.post("http://shipple.herokuapp.com/usps/", headers: { 'Content-Type' => 'application/json' }, body: params)
+
+  def ship
+    packages = set_packages
+    origin = set_origin
+    destination = set_destination
+
+    service = case params[:service]
+    when 'ups' then ups_credentials
+    when 'usps' then usps_credentials
+    end
```

Comment:
Again, style guide preferences, but this compact syntax makes this harder to read.

Separately, though, you do not have any error handling for what happens if params[:service] is *not* one of these 2 options.